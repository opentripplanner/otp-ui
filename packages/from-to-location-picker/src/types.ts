// eslint-disable-next-line prettier/prettier
import type { Location } from "@opentripplanner/types";

export type FromToPickerProps = {
  /**
   * Specifies the label to be rendered, or if set to true, renders the default label "Plan a trip:".
   */
  label?: boolean | React.ReactElement | string;
  /**
   * A specific location to associate with this. This is only used when combined
   * with the setLocation prop.
   */
  location?: Location;
  /**
   * Triggered when the user clicks on the "from" button.
   */
  onFromClick?: () => void;
  /**
   * Triggered when the user clicks on the "to" button.
   */
  onToClick?: () => void;
  /**
   * Triggered when the user clicks either the "from" or "to" button and there
   * are no from/to specific handler functions defined as props.
   *
   * Passes an argument as follows:
   * { locationType: "from/to", location, reverseGeocode: false }
   */
  setLocation?: (_: {
    locationType: string,
    // eslint-disable-next-line @typescript-eslint/no-shadow
    location: Location,
    reverseGeocode: boolean
  }) => void;
  /**
   * Determines whether icons are shown on the "from" and "to" buttons.
   */
  showIcons?: boolean;
};
