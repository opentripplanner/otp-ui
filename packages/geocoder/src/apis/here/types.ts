// Prettier does not support typescript annotation
// eslint-disable-next-line prettier/prettier
import type { LonLatInput } from "@conveyal/lonlat"

export interface HereResponse {
  items: Item[];
  point?: LonLatInput; // used in reverse response
}

export type Rect = {
  maxLat: number;
  maxLon: number;
  minLat: number;
  minLon: number;
};

export type Boundary = {
  country?: string;
  rect?: Rect;
};

export interface Item {
  access: Position[];
  address: Address;
  categories: Category[];
  distance: number;
  foodTypes: Category[];
  id: string;
  position: Position;
  resultType: string;
  scoring: Scoring;
  title: string;
}

export interface Position {
  lat: number;
  lng: number;
}

export interface Address {
  city: string;
  countryCode: string;
  countryName: string;
  county: string;
  district: string;
  houseNumber: string;
  label: string;
  postalCode: string;
  state: string;
  stateCode: string;
  street: string;
}

export interface Category {
  id: string;
  name: string;
  primary?: boolean;
}

export interface Scoring {
  fieldScore: FieldScore;
  queryScore: number;
}

export interface FieldScore {
  district: number;
  placeName: number;
}
