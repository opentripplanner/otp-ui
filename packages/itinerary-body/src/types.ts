import { FunctionComponent } from "react";

import {
  Config,
  FareProductSelector,
  GradationMap,
  Itinerary,
  Leg,
  LegIconComponent,
  Place,
  Stop,
  TransitOperator
} from "@opentripplanner/types";

export interface RouteDescriptionProps {
  /** Contains details about leg object that is being displayed */
  leg: Leg;
  /** A component class that is used to render icons for legs of an itinerary */
  LegIcon: LegIconComponent;
  /** The transit operator associated with the route if available */
  transitOperator: TransitOperator;
}

export interface RouteDescriptionFooterProps {
  /* Contains details about the leg object */
  leg?: Leg;
  /** Method for navigating to arrival vehicle if available */
  onClick?: () => void;
  /** Number of minutes for the arrival time */
  waitMinutes?: number;
}

export type ToRouteAbbreviationFunction = (route: string | number) => string;

export interface LegDestination {
  /** whether this place row represents the destination */
  isDestination: boolean;
  /** Contains details about leg object that is being displayed */
  leg: Leg;
}

export type TimeColumnContentProps = LegDestination;

/**
 * Shared props for various components that render leg data.
 */
export interface LegSharedProps extends LegDestination {
  /** Contains details about the leg object prior to the current one */
  lastLeg?: Leg;
  /** The index value of this specific leg within the itinerary */
  legIndex: number;
}

export interface LineColumnContentProps extends LegSharedProps {
  /** Whether this leg is an interlined-transit leg */
  interline: boolean;
  LegIcon: LegIconComponent;
  /** Converts a route's ID to its accepted badge abbreviation */
  toRouteAbbreviation: ToRouteAbbreviationFunction;
}

export interface PlaceNameProps {
  config: Config;
  interline?: boolean;
  place: Place;
}

export type SetActiveLegFunction = (legIndex: number, leg: Leg) => void;

export type FrameLegFunction = (
  args: LegSharedProps & {
    place: Place;
  }
) => void;

export interface TripSection {
  fromIndex?: number;
  fromStopId?: string;
  toIndex?: number;
  toStopId?: string;
  tripId: string;
}

export type SetViewedTripFunction = (tripSection: TripSection) => void;

export interface TransitLegSubheaderProps {
  leg: Leg;
  onStopClick?: (stop: Stop) => void;
}

export interface TransitLegSummaryProps {
  leg: Leg;
  onClick?: () => void;
  stopsExpanded: boolean;
}

/**
 * Shared props for various components that render itinerary data.
 */
interface ItineraryBodySharedProps {
  /**
   * A mapping of accessibility score to color, icon, and text used
   * to override the default one shipped in AccessibilityLabel
   */
  accessibilityScoreGradationMap?: GradationMap;
  /**
   * A custom icon component inserted into the transit alert body component
   * within a transit leg, if this prop is not supplied a default icon is used
   */
  AlertBodyIcon?: FunctionComponent;
  /**
   * A custom icon component inserted into the transit alert toggle button
   * within a transit leg, if this prop is not supplied a default icon is used
   */
  AlertToggleIcon?: FunctionComponent;
  /** If true, alerts in a trip leg always open in a collapsed state. */
  alwaysCollapseAlerts: boolean;
  /**
   * Used for additional styling with styled components for example.
   */
  className?: string;
  /** Contains OTP configuration details. */
  config: Config;
  /**
   * Allows selection of a fare product type for display in the itinerary body.
   * When fare leg information is available, it will be shown per-leg.
   * Example: regular, cash or regular, electronic.
   */
  defaultFareSelector?: FareProductSelector;
  /**
   * Should be either null or a legType. Indicates that a particular leg diagram
   * has been selected and is active.
   */
  diagramVisible?: Leg;
  /**
   * Called upon clicking the map icon on place headers. This function is sent a
   * single argument of an object with the keys as follow:
   * - `leg`: the leg clicked (can be null if the destination is clicked)
   * - `legIndex`: the index of the leg clicked (can be null if the destination
   *    is clicked)
   * - `isDestination`: if the place header that is clicked is the destination
   * - `place`: The place associated with the click event
   */
  frameLeg?: FrameLegFunction;
  /** A component class that is used to render icons for legs of an itinerary */
  LegIcon: LegIconComponent;
  /**
   * A slot for a component that can render the content in the line column.
   * This component is sent the following props:
   * - interline - whether this place is an interlined stop
   * - isDestination - whether this place is the destination
   * - lastLeg - the leg prior to the current leg
   * - leg - the current leg
   * - LegIcon - the LegIcon class used to render leg icons.
   * - legIndex - the current leg index
   * - toRouteAbbreviation - a function to help abbreviate route names
   */
  LineColumnContent: FunctionComponent<LineColumnContentProps>;
  /** Handler for when a Mapillary button is clicked. */
  mapillaryCallback?: (id: string) => void;
  /**
   * Mapillary key used to fetch imagery if available. Key can be obtained from
   * https://www.mapillary.com/dashboard/developers
   */
  mapillaryKey?: string;
  /**
   * A custom component for rendering the place name of legs.
   * The component is sent 3 props:
   * - config: the application config
   * - interline: whether this place is an interlined stop (a stop where a
   *   transit vehicle changes routes, but a rider can continue riding without
   *   deboarding)
   * - place: the particular place. Typically this is the from place, but it
   *   could also be the to place if it is the destination of the itinerary.
   */
  PlaceName: FunctionComponent<PlaceNameProps>;
  /**
   * A component to render the name of a route.
   *
   * The component is sent 3 props:
   * - leg: the itinerary leg with the transit information
   * - LegIcon: component that renders the icon for a leg.
   * - transitOperator: the transit operator associated with the route if available
   */
  RouteDescription: FunctionComponent<RouteDescriptionProps>;
  /**
   * A component to render the footer and contained elements
   * therein
   */
  RouteDescriptionFooter: FunctionComponent<RouteDescriptionFooterProps>;
  /** TODO: Routing Type is usually 'ITINERARY' but we should get more details on what this does */
  routingType?: string;
  /**
   * Sets the active leg and legIndex.
   * Called with 2 arguments: (legIndex, leg)
   */
  setActiveLeg: SetActiveLegFunction;
  /** Handler for when a leg diagram is selected. */
  setLegDiagram: (leg: Leg) => void;
  /** Fired when a user clicks on a view trip button of a transit leg */
  setViewedTrip: SetViewedTripFunction;
  /** If true, will show agency information in transit legs */
  showAgencyInfo?: boolean;
  /** If false, will not show the effective as of date/time text in the alert body */
  showAlertEffectiveDateTimeText?: boolean;
  /** Whether to prefix non transit legs with 'About' text */
  showApproximateAccessLegTravelTimes?: boolean;
  /** If true, will show the elevation profile for walk/bike legs */
  showElevationProfile?: boolean;
  /** If true will show the leg icon in the leg body */
  showLegIcon?: boolean;
  /** If true, will show the right column with the map button */
  showMapButtonColumn?: boolean;
  /** If true, shows the view trip button in transit leg bodies */
  showViewTripButton?: boolean;
  /**
   * A slot for a component that can render the content in the time column portion of ItineraryBody.
   * This component is sent the following props:
   * - isDestination - whether this place is the destination
   * - leg - the current leg
   */
  TimeColumnContent?: FunctionComponent<TimeColumnContentProps>;
  /** Converts a route's ID to its accepted badge abbreviation */
  toRouteAbbreviation?: ToRouteAbbreviationFunction;
  /**
   * An optional custom component for rendering a subheader on transit legs.
   * * The component is sent the following props:
   * - leg: the transit leg
   * - onStopClick: triggered when the optional stop icon is clicked.
   */
  TransitLegSubheader?: FunctionComponent<TransitLegSubheaderProps>;
  /**
   * A custom component for rendering the summary of a transit leg.
   * The component is sent 2 props:
   * - leg: the transit leg
   * - stopsExpanded: whether the intermediate stop display is currently expanded
   */
  TransitLegSummary: FunctionComponent<TransitLegSummaryProps>;
}

export interface PlaceRowProps
  extends ItineraryBodySharedProps,
    LegSharedProps {
  /** Indicates whether this leg directly follows a transit leg */
  followsTransit?: boolean;
}

export interface ItineraryBodyProps extends ItineraryBodySharedProps {
  /** Itinerary that the user has selected to view, contains multiple legs */
  itinerary: Itinerary;
}
