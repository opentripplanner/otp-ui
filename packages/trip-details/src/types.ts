// Prettier does not recognize the import type syntax.
// eslint-disable-next-line prettier/prettier
import type { 
  FareProductSelector, 
  Itinerary, 
  Leg, 
  MassUnitOption,
  Money
 } from "@opentripplanner/types";

export interface TimeActiveDetailsProps {
  bikeMinutes: number;
  minutesActive?: number;
  walkMinutes: number;
}

export interface CO2ConfigType {
  carbonIntensity?: Record<string, number>;
  units?: MassUnitOption;
  enabled?: boolean;
}

export interface DepartureDetailsProps {
  departureDate: Date;
}

/**
 * This is the interface used to define the layout for a particular fare table.
 * The table with be rendered with the columns defined here,
 * with each row being an individual transit leg from the itinerary.
 */
export interface FareTableLayout {
  cols: (FareProductSelector & {
    columnHeaderKey: string;
  })[]
  headerKey: string;
}
/**
 * Interface containing the lgs and the layout of the fare table.
 */
export interface FareLegTableProps {
  layout?: FareTableLayout[];
  legs: Leg[];
}

// Total fare amount corresponding to a fare key
export type FareTotals = (FareProductSelector & { price: Money })[]

export interface FareDetailsProps {
  legs: Leg[];
  maxTNCFare: number;
  minTNCFare: number;
}


export interface TripDetailsProps {
  /**
   * Used for additional styling with styled components for example.
   */
  className?: string;
    /**
   * Object containing the COâ‚‚ config.
   */
  co2Config?: CO2ConfigType;
  /**
   * Determines which transit fare should be displayed by default, should there be multiple transit fare types.
   */
  defaultFareType?: { headerKey: string } & FareProductSelector;
  /**
   * Slot for a custom component to render the expandable section for departure.
   */
  DepartureDetails?: React.ElementType<DepartureDetailsProps>;
  /**
   * If this is set to true, a row will be added to the trip details displaying how
   * many minutes in total the user will spend walking or biking.
   */
  displayTimeActive?: boolean;
  /**
   * Slot for a custom component to render the expandable section for fares.
   */
  FareDetails?: React.ElementType<FareDetailsProps>;
  /**
   * Column and table configuration for fare details/fare by leg table.
   */
  fareDetailsLayout?: FareTableLayout[];
  /**
   * Mapping between fare keys and human-readable names for them.
   */
  fareKeyNameMap?: {
    [name: string]: string;
  };
  /**
   * Itinerary that the user has selected to view, contains multiple legs.
   */
  itinerary: Itinerary;
    /**
   * Slot for a custom component to render the expandable section for time active.
   */
  TimeActiveDetails?: React.ElementType<TimeActiveDetailsProps>;
}