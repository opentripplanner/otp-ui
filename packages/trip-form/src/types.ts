import { FunctionComponent, ReactNode } from "react";
import { StyledComponent } from "styled-components";

export type QueryParams = Record<string, string>;

/**
 * Describes the parameters for the onQueryParamChange event.
 */
export interface QueryParamChangeEvent {
  [key: string]: string | number | boolean;
}

/**
 * Describes the text/value pair for each query parameter option.
 */
export interface QueryParamOptions {
  text: string;
  value: string | number | boolean;
}

/**
 * Describes full details of a configured/supported travel mode provider.
 */
// FIXME: merge with TripOptions ModeOption.
export interface FullModeOption {
  company?: string;
  image?: string;
  hidden?: boolean;
  label: string;
  mode: string;
  url?: string;
}

/**
 * Describes a configured/supported travel mode provider.
 */
// FIXME: merge with TripOptions ModeOption.
export type ModeOption = string | FullModeOption;

/**
 * Describes the travel modes available for the GeneralSettings component.
 */
export interface ConfiguredModes {
  accessModes?: ModeOption[];
  bicycleModes?: ModeOption[];
  exclusiveModes?: ModeOption[];
  micromobilityModes?: ModeOption[];
  transitModes?: ModeOption[];
}

/**
 * Describes a supported transportation company.
 */
export interface ConfiguredCompany {
  /**
   * The id of the company. This is typically in all-caps.
   */
  id: string;
  /**
   * A human readable text value that can be displayed to users.
   */
  label: string;
  /**
   * A comma-separated list of applicable modes of travel that the company
   * offers.
   */
  modes: string;
}

/**
 * Describes a custom query parameter message.
 * TODO: Move to core-utils once that package supports TypeScript.
 */
export interface CustomQueryParameter {
  label?: ReactNode;
  options?:
    | QueryParamOptions[]
    | ((queryParams: QueryParams) => QueryParamOptions[]);
}

/**
 * Describes a set of custom query parameter messages.
 * TODO: Move to core-utils once that package supports TypeScript.
 */
export type CustomQueryParameters = Record<string, CustomQueryParameter>;

/**
 * Mode/SubmodeSelector options
 */
// FIXME: merge with TripOptions ModeOption.
export interface ModeSelectorOption {
  id: string;
  selected?: boolean;
  showTitle?: boolean;
  text: ReactNode;
  title?: string;
}

/**
 * Describes the set of options passed to the ModeSelector component.
 */
export interface ModeSelectorOptionSet {
  primary?: ModeSelectorOption;
  secondary?: ModeSelectorOption[];
  tertiary?: ModeSelectorOption[];
}

type SimpleStyledDiv = StyledComponent<"div", any>;

export type DateTimeSelectorAndSubComponents = SimpleStyledDiv & {
  DateTimeRow?: SimpleStyledDiv;
  DepartureRow?: SimpleStyledDiv;
};

export type ModeSelectorAndSubComponents = SimpleStyledDiv & {
  MainRow?: SimpleStyledDiv;
  SecondaryRow?: SimpleStyledDiv;
  TertiaryRow?: SimpleStyledDiv;
};

export type SubmodeSelectorAndSubComponents = SimpleStyledDiv & {
  InlineRow?: SimpleStyledDiv;
  Row?: SimpleStyledDiv;
};

export type ModeButtonAndSubComponents = SimpleStyledDiv & {
  Button?: StyledComponent<"button", any>;
  Title?: SimpleStyledDiv;
};

// FIXME: move to icons package when converting that to TypeScript.
export type ModeIconType = FunctionComponent<{ mode?: string }>;
