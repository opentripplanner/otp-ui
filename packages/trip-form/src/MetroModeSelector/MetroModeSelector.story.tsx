import { ModeButtonDefinition } from "@opentripplanner/types";
import React, { ReactElement, useState } from "react";
import * as Core from "..";
import { QueryParamChangeEvent } from "../types";
import {
  addSettingsToButton,
  extractModeSettingDefaultsToObject,
  populateSettingWithValue,
  setModeButtonEnabled
} from "./utils";
import {
  modeSettingDefinitionsWithDropdown,
  getIcon,
  defaultModeButtonDefinitions
} from "../__mocks__/mode-selector-buttons";

const initialState = {
  enabledModeButtons: ["transit"],
  modeSettingValues: {}
};

function pipe<T>(...fns: Array<(arg: T) => T>) {
  return (value: T) => fns.reduce((acc, fn) => fn(acc), value);
}

const MetroModeSelectorComponent = ({
  fillModeIcons,
  modeButtonDefinitions,
  onSetModeSettingValue,
  onToggleModeButton
}: {
  fillModeIcons?: boolean;
  modeButtonDefinitions: ModeButtonDefinition[];
  onSetModeSettingValue: (event: QueryParamChangeEvent) => void;
  onToggleModeButton: (key: string, newState: boolean) => void;
}): ReactElement => {
  const [modeSettingValues, setModeSettingValues] = useState({});
  const modeSettingValuesWithDefaults = {
    ...extractModeSettingDefaultsToObject(modeSettingDefinitionsWithDropdown),
    ...initialState.modeSettingValues,
    ...modeSettingValues
  };

  const [activeModeButtonKeys, setModeButtonKeys] = useState(
    initialState.enabledModeButtons
  );

  const addIconToModeSetting = msd => ({
    ...msd,
    icon: getIcon(msd.iconName)
  });

  const processedModeSettings = modeSettingDefinitionsWithDropdown.map(
    pipe(
      addIconToModeSetting,
      populateSettingWithValue(modeSettingValuesWithDefaults)
    )
  );

  const processedModeButtons = modeButtonDefinitions.map(
    pipe(
      addSettingsToButton(processedModeSettings),
      setModeButtonEnabled(activeModeButtonKeys)
    )
  );

  const toggleModeButtonAction = (key: string, newState: boolean) => {
    if (newState) {
      setModeButtonKeys([...activeModeButtonKeys, key]);
    } else {
      setModeButtonKeys(activeModeButtonKeys.filter(button => button !== key));
    }
    // Storybook Action:
    onToggleModeButton(key, newState);
  };

  const setModeSettingValueAction = (event: QueryParamChangeEvent) => {
    setModeSettingValues({ ...modeSettingValues, ...event });
    // Storybook Action:
    onSetModeSettingValue(event);
  };

  return (
    <Core.MetroModeSelector
      fillModeIcons={fillModeIcons}
      label="Select a transit mode"
      modeButtons={processedModeButtons}
      onSettingsUpdate={setModeSettingValueAction}
      onToggleModeButton={toggleModeButtonAction}
    />
  );
};

const Template = (args: {
  fillModeIcons?: boolean;
  onSetModeSettingValue: (event: QueryParamChangeEvent) => void;
  onToggleModeButton: (key: string, newState: boolean) => void;
}): ReactElement => (
  <MetroModeSelectorComponent
    modeButtonDefinitions={defaultModeButtonDefinitions}
    // eslint-disable-next-line react/jsx-props-no-spreading
    {...args}
  />
);

export default {
  argTypes: {
    fillModeIcons: { control: "boolean" },
    onSetModeSettingValue: { action: "set mode setting value" },
    onToggleModeButton: { action: "toggle button" }
  },
  component: MetroModeSelectorComponent,
  title: "Trip Form Components/Metro Mode Selector"
};

export const MetroModeSelector = Template.bind({});

const a11yOverrideParameters = {
  a11y: { config: { rules: [{ id: "label", reviewOnFail: true }] } }
};
MetroModeSelector.parameters = a11yOverrideParameters;
