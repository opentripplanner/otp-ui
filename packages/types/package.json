{
  "name": "@opentripplanner/types",
  "version": "6.1.0",
  "description": "TypeScript types used across multiple OTP-UI packages",
  "main": "lib/index.js",
  "types": "lib/index.d.ts",
  "repository": "https://github.com/opentripplanner/otp-ui.git",
  "author": "Binh Dam",
  "license": "MIT",
  "private": false,
  "scripts": {
    "tsc": "tsc"
  }
}
