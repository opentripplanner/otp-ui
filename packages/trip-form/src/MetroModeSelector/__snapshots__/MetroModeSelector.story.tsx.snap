// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Trip Form Components/Metro Mode Selector MetroModeSelector smoke-test 1`] = `
<fieldset class="MetroModeSelector__ModeBar-sc-n9bu6a-1 ezAten metro-mode-selector">
  <legend>
    Select a transit mode
  </legend>
  <span class="MetroModeSelector__ModeButtonWrapper-sc-n9bu6a-2 klXmgn">
    <input aria-label="Transit"
           id="metro-mode-selector-mode-transit"
           type="checkbox"
           checked
    >
    <label for="metro-mode-selector-mode-transit"
           title="Transit"
    >
      <svg viewbox="0 0 512 512"
           height="32"
           width="32"
           aria-hidden="true"
           focusable="false"
           fill="currentColor"
           xmlns="http://www.w3.org/2000/svg"
           class="StyledIconBase-sc-ea9ulj-0 kcHuPT"
      >
        <path fill="currentColor"
              d="M256 0c134.4 0 224 35.2 224 80v48c17.7 0 32 14.3 32 32v64c0 17.7-14.3 32-32 32v160c0 17.7-14.3 32-32 32v32c0 17.7-14.3 32-32 32h-32c-17.7 0-32-14.3-32-32v-32H160v32c0 17.7-14.3 32-32 32H96c-17.7 0-32-14.3-32-32v-32c-17.7 0-32-14.3-32-32V256c-17.7 0-32-14.3-32-32v-64c0-17.7 14.3-32 32-32V80c0-44.8 89.6-80 224-80zM96 160v96c0 17.7 14.3 32 32 32h112V128H128c-17.7 0-32 14.3-32 32zm176 128h112c17.7 0 32-14.3 32-32v-96c0-17.7-14.3-32-32-32H272v160zM112 400c17.7 0 32-14.3 32-32s-14.3-32-32-32-32 14.3-32 32 14.3 32 32 32zm288 0c17.7 0 32-14.3 32-32s-14.3-32-32-32-32 14.3-32 32 14.3 32 32 32zM352 80c0-8.8-7.2-16-16-16H176c-8.8 0-16 7.2-16 16s7.2 16 16 16h160c8.8 0 16-7.2 16-16z"
        >
        </path>
      </svg>
      <span class="MetroModeSelector__InvisibleA11yLabel-sc-n9bu6a-0 gcxxlD">
        Transit
      </span>
    </label>
  </span>
  <span class="MetroModeSelector__ModeButtonWrapper-sc-n9bu6a-2 klXmgn">
    <input aria-label="Walk"
           id="metro-mode-selector-mode-walk"
           type="checkbox"
    >
    <label for="metro-mode-selector-mode-walk"
           title="Walk"
    >
      <svg viewbox="0 0 320 512"
           height="32"
           width="32"
           aria-hidden="true"
           focusable="false"
           fill="currentColor"
           xmlns="http://www.w3.org/2000/svg"
           class="StyledIconBase-sc-ea9ulj-0 kcHuPT"
      >
        <path fill="currentColor"
              d="M256 48c0 26.5-21.5 48-48 48s-48-21.5-48-48 21.5-48 48-48 48 21.5 48 48zM126.5 199.3c-1 .4-1.9.8-2.9 1.2l-8 3.5c-16.4 7.3-29 21.2-34.7 38.2l-2.6 7.8c-5.6 16.8-23.7 25.8-40.5 20.2S12 246.5 17.6 229.7l2.6-7.8c11.4-34.1 36.6-61.9 69.4-76.5l8-3.5c20.8-9.2 43.3-14 66.1-14 44.6 0 84.8 26.8 101.9 67.9l15.4 36.9 21.4 10.7c15.8 7.9 22.2 27.1 14.3 42.9s-27.1 22.2-42.9 14.3L247 287.3c-10.3-5.2-18.4-13.8-22.8-24.5l-9.6-23-19.3 65.5 49.5 54c5.4 5.9 9.2 13 11.2 20.8l23 92.1c4.3 17.1-6.1 34.5-23.3 38.8s-34.5-6.1-38.8-23.3l-22-88.1-70.7-77.1c-14.8-16.1-20.3-38.6-14.7-59.7l16.9-63.5zM68.7 398l25-62.4c2.1 3 4.5 5.8 7 8.6l40.7 44.4-14.5 36.2c-2.4 6-6 11.5-10.6 16.1l-61.7 61.7c-12.5 12.5-32.8 12.5-45.3 0s-12.5-32.8 0-45.3L68.7 398z"
        >
        </path>
      </svg>
      <span class="MetroModeSelector__InvisibleA11yLabel-sc-n9bu6a-0 gcxxlD">
        Walk
      </span>
    </label>
  </span>
  <span class="MetroModeSelector__ModeButtonWrapper-sc-n9bu6a-2 klXmgn">
    <input aria-label="Bike"
           id="metro-mode-selector-mode-bicycle"
           type="checkbox"
    >
    <label for="metro-mode-selector-mode-bicycle"
           title="Bike"
    >
      <svg xmlns="http://www.w3.org/2000/svg"
           version="1.1"
           viewbox="0 0 100 61.3"
           height="100%"
           width="100%"
           aria-hidden="true"
           size="32"
      >
        <path d="M0,41.6C0,52.3,8.7,61,19.4,61c9.4,0,17.2-6.7,19-15.5c0.1,0,0.1,0,0.2,0c15.3-0.6,4.6,4.2,33-25.7c0.5,1.5,0.9,2.9,1.4,4.3  c-6.9,3-11.8,9.8-11.8,17.9c0,10.7,8.7,19.4,19.4,19.4c10.7,0,19.4-8.7,19.4-19.4s-8.7-19.4-19.4-19.4c-1.2,0-2.4,0.1-3.6,0.3  C69.5-1.5,71.9,0.1,57.9,0v3.9c10.2,0,8.9-0.6,11.2,7.3H35.3L34,7.5c7.6-1.7,9.7-1.7,8.2-3.7H25.1v3.8l4.5,0.1l1.3,3.6  c-2.1,4.6-3.8,8.5-5.3,11.9c-2-0.7-4-1-6.2-1C8.7,22.2,0,30.9,0,41.6z M74.4,28.4c4.8,15.4,4.4,15.1,6.3,15.3c1.1-0.1,1.9-0.9,1.9-2  c0-1.3,0.5,0.4-4.2-14.5c0.7-0.1,1.4-0.2,2.2-0.2c8.2,0,14.9,6.7,14.9,14.9s-6.7,14.9-14.9,14.9s-14.9-6.7-14.9-14.9  C65.7,35.9,69.3,30.7,74.4,28.4z M37,15.3l33,0.1L46.6,41L37,15.3z M29.4,24.9l3.6-8.1l9.4,24.6l-3.6,0C38.8,34.5,35,28.3,29.4,24.9  z M34.3,41.5l-12.3,0l5.6-12.4C31.6,31.8,34.2,36.3,34.3,41.5z M4.5,41.6c0-8.2,6.7-14.9,14.9-14.9c1.5,0,3,0.2,4.3,0.6  c-6.7,14.9-7,15.4-7,16.2c0.2,2.7,1.6,1.9,17.1,1.9c-1.7,6.3-7.5,11-14.3,11C11.2,56.4,4.5,49.8,4.5,41.6z">
        </path>
      </svg>
      <span class="MetroModeSelector__InvisibleA11yLabel-sc-n9bu6a-0 gcxxlD">
        Bike
      </span>
    </label>
  </span>
  <span class="MetroModeSelector__ModeButtonWrapper-sc-n9bu6a-2 klXmgn">
    <input aria-label="Drive"
           id="metro-mode-selector-mode-car"
           type="checkbox"
    >
    <label for="metro-mode-selector-mode-car"
           title="Drive"
    >
      <svg viewbox="0 0 512 512"
           height="32"
           width="32"
           aria-hidden="true"
           focusable="false"
           fill="currentColor"
           xmlns="http://www.w3.org/2000/svg"
           class="StyledIconBase-sc-ea9ulj-0 kcHuPT"
      >
        <path fill="currentColor"
              d="M135.2 117.4 109.1 192h293.8l-26.1-74.6c-4.5-12.8-16.6-21.4-30.2-21.4H165.4c-13.6 0-25.7 8.6-30.2 21.4zm-95.6 79.4L74.8 96.3C88.3 57.8 124.6 32 165.4 32h181.2c40.8 0 77.1 25.8 90.6 64.3l35.2 100.5c23.2 9.6 39.6 32.5 39.6 59.2v192c0 17.7-14.3 32-32 32h-32c-17.7 0-32-14.3-32-32v-48H96v48c0 17.7-14.3 32-32 32H32c-17.7 0-32-14.3-32-32V256c0-26.7 16.4-49.6 39.6-59.2zM128 288c0-17.7-14.3-32-32-32s-32 14.3-32 32 14.3 32 32 32 32-14.3 32-32zm288 32c17.7 0 32-14.3 32-32s-14.3-32-32-32-32 14.3-32 32 14.3 32 32 32z"
        >
        </path>
      </svg>
      <span class="MetroModeSelector__InvisibleA11yLabel-sc-n9bu6a-0 gcxxlD">
        Drive
      </span>
    </label>
  </span>
</fieldset>
`;
